modules/exploits/windows/http/magicinfo_traversal.rb

```diff
@@ -0,0 +1,119 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Samsung MagicINFO 9 Server Remote Code Execution (CVE-2024-7399)',
+        'Description' => %q{
+          Remote Code Execution in Samsung MagicINFO 9 Server <= 21.1050.0.
+          Remote code execution can be obtained by exploiting the path traversal vulnerability (CVE-2024-7399) in the SWUpdateFileUploader servlet,
+          which can be queried by an unauthenticated user to upload a JSP shell.
+          By default, the application listens on TCP ports 7001 (HTTP) and 7002 (HTTPS) on all network interfaces and runs in the context of NT AUTHORITY\SYSTEM.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'SSD Secure Disclosure' # Discovery and PoC
+        ],
+        'References' => [
+          [ 'URL', 'https://ssd-disclosure.com/ssd-advisory-samsung-magicinfo-unauthenticated-rce/'],
+          [ 'URL', 'https://security.samsungtv.com/securityUpdates'],
+          [ 'CVE', '2024-7399'] # SVE-2024-50018
+        ],
+        'DisclosureDate' => '2025-04-30',
+        'DefaultOptions' => {
+          'RPORT' => 7002,
+          'SSL' => 'True'
+        },
+        'Platform' => [ 'windows' ],
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Java Server Page', {
+              'Platform' => %w[win],
+              'Arch' => ARCH_JAVA
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
```

Comment:
This module gets a session as `system`.

```suggestion
        'DefaultTarget' => 0,
        'Privileged' => true,
```

Reply:
Updated, thank you.